src/opensearchDatasource.test.ts

```diff
@@ -128,1482 +113,114 @@ describe('OpenSearchDatasource', function (this: any) {
     });
 
     it('should translate index pattern to current day', () => {
-      let requestOptions: any;
-      datasourceRequestMock.mockImplementation((options) => {
-        requestOptions = options;
+      let resourcePath: any;
+      ctx.ds.getResource = jest.fn().mockImplementation((path: string) => {
+        resourcePath = path;
         return Promise.resolve({ data: {} });
       });
 
-      ctx.ds.testDatasource();
-
-      const today = toUtc().format('YYYY.MM.DD');
-      expect(requestOptions.url).toBe(`${OPENSEARCH_MOCK_URL}/asd-${today}/_mapping`);
-    });
-  });
-
-  describe('When issuing metric query with interval pattern', () => {
-    let requestOptions: any, parts: any, header: any, query: any, result: any;
-
-    beforeEach(async () => {
-      createDatasource({
-        url: OPENSEARCH_MOCK_URL,
-        jsonData: {
-          database: '[asd-]YYYY.MM.DD',
-          interval: 'Daily',
-          version: '1.0.0',
-        } as OpenSearchOptions,
-      } as DataSourceInstanceSettings<OpenSearchOptions>);
-
-      datasourceRequestMock.mockImplementation((options) => {
-        requestOptions = options;
-        return Promise.resolve({
-          data: {
-            responses: [
-              {
-                aggregations: {
-                  '1': {
-                    buckets: [
-                      {
-                        doc_count: 10,
-                        key: 1000,
-                      },
-                    ],
-                  },
-                },
-              },
-            ],
-          },
-        });
-      });
-
-      query = {
-        range: {
-          from: toUtc([2015, 4, 30, 10]),
-          to: toUtc([2015, 5, 1, 10]),
-        },
-        targets: [
-          {
-            alias: '$varAlias',
-            bucketAggs: [{ type: 'date_histogram', field: '@timestamp', id: '1' }],
-            metrics: [{ type: 'count', id: '1' }],
-            query: 'escape\\:test',
-          },
-        ],
-      };
-
-      result = await ctx.ds.query(query).toPromise();
-
-      parts = requestOptions.data.split('\n');
-      header = JSON.parse(parts[0]);
-    });
-
-    it('should translate index pattern to current day', () => {
```

Comment:
I notice we only kept the `'should not resolve the variable in the original alias field in the query'`, are the other tests just not applicable anymore?

Reply:
Since the others in the describe block test query building, which we do on the backend now, I don't think so